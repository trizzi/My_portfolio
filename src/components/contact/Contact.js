import React from 'react';
import contact from '../../assets/Contact.svg';
import text from '../../assets/contact_me_text.svg';

const Contact = () => {
	return (
		<div
			name='section5'
			className='flex text-white w-full'>
			<div className='hidden md:block md:visible w-1/2 bg-black p-32'>
				<img
					src={text}
					alt='Contact-me-text'
				/>
			</div>
			<div className='relative sm:w-1/2 bg-black p-12 lg:p-24'>
				<img
					src={contact}
					alt='images'
				/>
				<div className='absolute top-1/4 left-[9%] transform hover:scale-110 transition-transform duration-300 bg-green-500 rounded-full p-1 md:p-2'>
					<i className='fa-brands fa-whatsapp text-white text-2xl md:text-3xl'></i>
				</div>

				<div className='absolute top-[18%] left-[60%] transform -translate-x-1/2 -translate-y-1/2 hover:rotate-45 transition-transform duration-300 bg-blue-500 rounded-full p-1 md:p-2'>
					<a href='https://www.linkedin.com/in/patrick-akhamiogu-8750b6177/'>
						<i className='fa-brands fa-linkedin text-white text-2xl '></i>
					</a>
				</div>

				<div className='absolute top-[78%] left-[30%] transform hover:animate-bounce bg-black rounded-full p-1 md:p-2'>
					<a href='https://www.github.com/trizzi'>
						<i className='fa-brands fa-github text-white text-2xl md:text-3xl'></i>
					</a>
				</div>

				<div className='absolute top-1/3 left-[84%] md:left-[71%] transform hover:animate-bounce bg-black rounded-full p-1 md:p-2'>
					<a href='https://medium.com/@patrickakhamiogu'>
						<i className='fa-brands fa-medium text-white text-2xl'></i>
					</a>
				</div>

				<div className='absolute top-1/2 left-[10%] transform hover:flip-horizontal duration-300 bg-white rounded-full p-1 md:p-2'>
					<a href='patrickakhamiogu@gmail.com'>
						<i className='fa-brands fa-google text-red-400 text-2xl md:text-3xl'></i>
					</a>
				</div>

				<div className='absolute top-[14%] left-1/3 transform hover:flip-horizontal duration-300 bg-blue-500 rounded-full p-1 md:p-2'>
					<a href='https://www.facebook.com/patrick.akhamiogu'>
						<i className='fa-brands fa-facebook text-white text-2xl md:text-3xl'></i>
					</a>
				</div>

				<div className='absolute top-2/3 left-[84%] md:left-[71%] transform hover:animate-bounce bg-sky-500 rounded-full p-1 md:p-2'>
					<a href='https://twitter.com/Trizzi_ric'>
						<i className='fa-brands fa-twitter text-white text-2xl md:text-3xl'></i>
					</a>
				</div>
			</div>
		</div>
	);
};

export default Contact;
