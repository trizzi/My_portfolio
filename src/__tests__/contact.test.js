import React from 'react';
import { render } from '@testing-library/react';
import Contact from '../components/contact/Contact';
import '@testing-library/jest-dom/extend-expect';

import { ThemeContextProvider } from '../context/ThemeContext';

jest.useFakeTimers();

test('renders contact component with all social media links', () => {
	const { container } = render(
		<ThemeContextProvider>
			<Contact />
		</ThemeContextProvider>,
	);

	// Check if the component renders with all social media links
	const telegramLink = document.querySelector('a[href="https://t.me/PJ923"]');
	expect(telegramLink).toBeInTheDocument();

	const linkedinLink = document.querySelector(
		'a[href="https://www.linkedin.com/in/patrick-akhamiogu-8750b6177/"]',
	);
	expect(linkedinLink).toBeInTheDocument();

	const githubLink = document.querySelector(
		'a[href="https://www.github.com/trizzi"]',
	);
	expect(githubLink).toBeInTheDocument();

	const mediumLink = document.querySelector(
		'a[href="https://medium.com/@patrickakhamiogu"]',
	);
	expect(mediumLink).toBeInTheDocument();

	const gmailLink = document.querySelector('a[href^="mailto:"]');
	expect(gmailLink).toBeInTheDocument();

	const facebookLink = document.querySelector(
		'a[href="https://www.facebook.com/patrick.akhamiogu"]',
	);
	expect(facebookLink).toBeInTheDocument();

	const twitterLink = document.querySelector(
		'a[href="https://twitter.com/Trizzi_ric"]',
	);
	expect(twitterLink).toBeInTheDocument();

	// Advance timers to trigger the visibility of each link
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);
	jest.advanceTimersByTime(1000);

	expect(container).toMatchSnapshot();

	// // Check if all links become visible after a certain timeout
	// expect(telegramLink).toHaveStyle('opacity: 1');
	// expect(linkedinLink).toHaveStyle('opacity: 1');
	// expect(githubLink).toHaveStyle('opacity: 1');
	// expect(mediumLink).toHaveStyle('opacity: 1');
	// expect(gmailLink).toHaveStyle('opacity: 1');
	// expect(facebookLink).toHaveStyle('opacity: 1');
	// expect(twitterLink).toHaveStyle('opacity: 1');
});
