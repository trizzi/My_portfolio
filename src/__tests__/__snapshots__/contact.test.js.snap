// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders contact component with all social media links 1`] = `
<div>
  <div
    class="flex flex-wrap md:flex-nowrap text-white w-full"
    name="section6"
  >
    <div
      class="w-full bg-black p-12 md:p-32"
    >
      <img
        alt="Contact-me-text"
        src="book_appointment.svg"
      />
      <a>
        <button
          class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
          style="background-color: rgb(153, 50, 204); color: rgb(255, 255, 255); border: 2px solid #9932cc; opacity: 0;"
          tabindex="0"
        >
          Book Appointment
        </button>
      </a>
    </div>
    <div
      class="relative w-full bg-black p-12 lg:p-24"
    >
      <img
        alt="images"
        src="Contact.svg"
      />
      <div
        class="absolute top-1/4 left-[9%] transform hover:scale-110 transition-transform  transition-opacity duration-300 bg-blue-500 rounded-full p-1 md:p-2"
        data-testid="telegram-link"
      >
        <a
          href="https://t.me/PJ923"
        >
          <i
            class="fa-brands fa-telegram text-white text-2xl md:text-3xl"
          />
        </a>
      </div>
      <div
        class="absolute top-[18%] left-[60%] transform -translate-x-1/2 -translate-y-1/2 hover:rotate-45 transition-transform  transition-opacity  duration-300 bg-blue-500 rounded-full p-1 md:p-2"
        data-testid="linkedin-link"
      >
        <a
          href="https://www.linkedin.com/in/patrick-akhamiogu-8750b6177/"
        >
          <i
            class="fa-brands fa-linkedin text-white text-2xl "
          />
        </a>
      </div>
      <div
        class="absolute top-[78%] left-[30%] transform   hover:animate-bounce bg-black rounded-full p-1 md:p-2"
        data-testid="github-link"
      >
        <a
          href="https://www.github.com/trizzi"
        >
          <i
            class="fa-brands fa-github text-white text-2xl md:text-3xl"
          />
        </a>
      </div>
      <div
        class="absolute top-1/3 left-[84%] md:left-[71%] transform   hover:animate-bounce bg-black rounded-full p-1 md:p-2"
        data-testid="medium-link"
      >
        <a
          href="https://medium.com/@patrickakhamiogu"
        >
          <i
            class="fa-brands fa-medium text-white text-2xl"
          />
        </a>
      </div>
      <div
        class="absolute top-1/2 left-[10%] transform  hover:flip-horizontal duration-300 bg-white rounded-full p-1 md:p-2"
        data-testid="gmail-link"
      >
        <a
          href="mailto:patrickakhamiogu@gmail.com"
        >
          <i
            class="fa-brands fa-google text-red-400 text-2xl md:text-3xl"
          />
        </a>
      </div>
      <div
        class="absolute top-[14%] left-1/3 transform  hover:flip-horizontal duration-300 bg-blue-500 rounded-full p-1 md:p-2"
        data-testid="facebook-link"
      >
        <a
          href="https://www.facebook.com/patrick.akhamiogu"
        >
          <i
            class="fa-brands fa-facebook text-white text-2xl md:text-3xl"
          />
        </a>
      </div>
      <div
        class="absolute top-2/3 left-[84%] md:left-[71%] transform   hover:animate-bounce bg-sky-500 rounded-full p-1 md:p-2"
        data-testid="twitter-link"
      >
        <a
          href="https://twitter.com/Trizzi_ric"
        >
          <i
            class="fa-brands fa-twitter text-white text-2xl md:text-3xl"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
